tests/functional/iac/test_iac_scan_ci.py

```diff
@@ -56,3 +56,44 @@ def test_ci_diff_no_vuln(
         file_content == _IAC_SINGLE_VULNERABILITY
     )
     assert expected_output in result.stdout
+
+
+def test_gitlab_previous_commit_sha_for_merged_results_pipelines(
+    tmp_path: Path,
+) -> None:
+    # GIVEN a remote repository
+    remote_repository = Repository.create(tmp_path / "remote", bare=True)
+    # AND a local clone
+    local_tmp_path = tmp_path / "local"
+    repository = Repository.clone(remote_repository.path, local_tmp_path)
+    repository.create_commit()
+    repository.push()
+
+    # AND a commit containing a file with or without a vulnerability
+    test_file = local_tmp_path / "test_file.tf"
+    test_file.write_text(_IAC_SINGLE_VULNERABILITY)
+    repository.add(test_file)
+    last_sha = repository.create_commit()
+
+    # AND an unstaged file with a vulnerability
+    iac_file = local_tmp_path / "iac_file.tf"
+    iac_file.write_text(_IAC_SINGLE_VULNERABILITY)
```

Comment:
I am not sure to understand the purpose of this file. In a CI, you shouldn't have a modified file, should you ?

Reply:
I've replaced it with a file pre-existing the push, which is more relevant. Thank you